python/lsst/daf/butler/core/utils.py

```diff
@@ -145,30 +146,38 @@ def doImport(pythonType):
     ------
     TypeError
         pythonType is not a `str`.
-    ValueError
-        pythonType can not be imported.
+    ModuleNotFoundError
+        No module in the supplied import string could be found.
+    ImportError
+        pythonType is found but can not be imported.
```

Comment:
I think it might be best to catch and re-raise any of these as `ImportError`; it's much more likely that the caller will want to catch anything (without having a too-general `except Exception` block) than try to distinguish between them.

Reply:
Yeah, using a hierarchy of exceptions is the best scenario.  If you can convert the `AttributeError` to `ImportError`, I do think that'd be best, even if that is a bit unlike `import` itself.